### PR TITLE
PRODTASKS-701 Code Coverage

### PR DESCRIPTION
**Pull request by dwalluck**

### CODE REVIEW

**Comment by dwalluck** [comment]
Does this need `'[ "$TRAVIS_OS_NAME" == "linux" ] && bash'` to prevent multiple uploads? I am not sure if codecov just takes the first one that finishes.

**Comment by dwalluck** [comment]
I don't think we'll see anything until the codecov support is merged into master. It looks like it successfully uploaded in the Travis logs.

File: pom.xml [hunk 1]
```diff
@@ -77,7 +77,7 @@
     <javaVersion>1.8</javaVersion>
     <javaTestVersion>${javaVersion}</javaTestVersion>
 
-    <mavenVersion>3.5.0</mavenVersion>
```
Comment (rnc) [inline pom.xml]: Why this change?

File: pom.xml [hunk 2]
```diff
@@ -95,44 +95,55 @@
       <dependency>
         <groupId>org.commonjava.maven.ext</groupId>
         <artifactId>pom-manipulation-common</artifactId>
-        <version>3.9-SNAPSHOT</version>
+        <version>${project.version}</version>
       </dependency>
       <dependency>
```
Comment (rnc) [inline pom.xml]: Why this change?

File: pom.xml (hunk 1 above)
Comment (dwalluck) [reply pom.xml]: No problem. I didn't mean to leave this in. I was just checking if the tests passed regardless of the version used.

File: pom.xml (hunk 2 above)
Comment (dwalluck) [reply pom.xml]: This was to make sure that these dependencies automatically keep the version in sync with the project version since these dependencies should always come from the project build itself and not an external maven repository.

**Comment by dwalluck** [comment]
> * What does this cover (i.e. coverage scope)?

I know that when the integration tests module is included, the missed instructions goes from about 12,000 to about 4,000.

File: pom.xml (hunk 2 above)
Comment (rnc) [reply pom.xml]: The release plugin handles that - see http://central.maven.org/maven2/org/commonjava/maven/ext/pom-manipulation-parent/3.8.1/pom-manipulation-parent-3.8.1.pom for example. Also using $project.version is really not advised (which is why PME inlines it when it encounters it). Please revert this.

File: pom.xml (hunk 1 above)
Comment (rnc) [reply pom.xml]: Ok

**Comment by rnc** [comment]
> > * What does this cover (i.e. coverage scope)?
> 
> I know that when the integration tests module is included, the number of missed instructions goes from about 12,000 to about 4,000.

I think you're misunderstanding me. As your PR stands - is this just covering unit tests or unit tests and integration tests? I would have expected to see e.g. jacoco goals of `prepare-agent-integration`?

**Comment by rnc** [comment]
> Does this need `'[ "$TRAVIS_OS_NAME" == "linux" ] && bash'` to prevent multiple uploads? I am not sure if codecov just takes the first one that finishes.

Can you elaborate please?

File: pom.xml (hunk 2 above)
Comment (dwalluck) [reply pom.xml]: > The release plugin handles that - see http://central.maven.org/maven2/org/commonjava/maven/ext/pom-manipulation-parent/3.8.1/pom-manipulation-parent-3.8.1.pom for example. Also using $project.version is really not advised (which is why PME inlines it when it encounters it). Please revert this.

This doesn't make a lot of sense to me as there are already five or more version properties defined in the pom, presumably for the same reason, to avoid errors and repeatedly hardcoding the same value over and over.

**Comment by dwalluck** [comment]
> Can you elaborate please?

If you look at the Travis CI logs, there's an upload of the coverage report for every build that finishes instead of one per pull request. So which one is it using? Just randomly the first build to finish? So, I was thinking how this could be limited to one report.

File: pom.xml (hunk 2 above)
Comment (rnc) [reply pom.xml]: Those aren't using project.version and, as I said, the release plugin handles transforming this. Its done on purpose ;-)

**Comment by rnc** [comment]
> > > * What does this cover (i.e. coverage scope)?
> > 
> > 
> > I know that when the integration tests module is included, the number of missed instructions goes from about 12,000 to about 4,000.
> 
> I think you're misunderstanding me. As your PR stands - is this just covering unit tests or unit tests and integration tests? I would have expected to see e.g. jacoco goals of `prepare-agent-integration`?

I think you missed this comment :-)
